lib/solarwinds_apm/sdk/trace_context_headers.rb

```diff
@@ -43,6 +43,10 @@ module TraceContextHeaders
       # * The headers with w3c tracecontext added, also modifies the headers arg if given
       #
       def add_tracecontext_headers(headers = {})
+        # make sure the header object can take string keys
+        # TODO maybe there is a better check?
+        return if headers.is_a?(Array)
```

Comment:
Under what circumstances would there be type confusion here? Only when user is using sdk, or also as result of auto instrumentation, or something else?

Reply:
now this method is applied in jobs as well
A job can just be an array of arguments with the worker knowing what to do, but then we cannot add trace info
if it is not an array it should be possible to add the headers 
I think it may be a good idea to catch exceptions in this method, because a job can be a thing that can't take key/values in which case we can't send the info. Ruby doesn't have a method to check if an obj takes key/values
I'll add the exception catching to part 2 / next PR